olive/evaluator/metric_config.py

```diff
@@ -29,17 +31,19 @@
 
 _type_to_user_config = {
     "latency": {
-        "dataloader_func": ConfigParam(type_=Union[Callable, str], is_object=True),
-        "inference_settings": ConfigParam(type_=dict),
         "io_bind": ConfigParam(type_=bool, default_value=False),
     },
     "accuracy": {
-        "dataloader_func": ConfigParam(type_=Union[Callable, str], is_object=True),
         "post_processing_func": ConfigParam(type_=Union[Callable, str], is_object=True),
-        "inference_settings": ConfigParam(type_=dict),
     },
     "custom": {
-        "evaluate_func": ConfigParam(type_=Union[Callable, str], required=True, is_object=True),
+        "evaluate_func": ConfigParam(type_=Union[Callable, str], required=False, is_object=True),
+        "evaluate_func_pytorch": ConfigParam(type_=Union[Callable, str], required=False, is_object=True),
+        "evaluate_func_onnx": ConfigParam(type_=Union[Callable, str], required=False, is_object=True),
+        "evaluate_func_snpe": ConfigParam(type_=Union[Callable, str], required=False, is_object=True),
+        "evaluate_func_openvino": ConfigParam(type_=Union[Callable, str], required=False, is_object=True),
```

Comment:
I suppose the evaluate_func should be enough for the functionality. what's the consideration that expose those function?

Reply:
In some cases, user needs to evaluate different type models. If we only have one custom eval func, it is not ideal for the user to evaluate all kinds of models in the same func.

Reply:
could you please clarify? I still think we don't need them.

Reply:
You can reference current `eval_accuracy` custom eval func in resnet example. There is an IF statement to distinguish different model type and run the inference: onnx model or pytorch model. So the concerns here are:
1. If there are multiple model types in a same workflow, but we only have one custom eval func for them, all different types model evaluation logic will be put in a same evaluation func.
2. As for custom eval function, I prefer to offer user a loaded model as an argument in the custom eval func. Instead of pass an additional arg like model type, having separate eval func for different model type will be cleaner.

Reply:
that's is user logic, user should have the full control about how to do the evaluation/inference from their point of view. 
Basically, we should provide the functionality support to allow user have full control of get the predicts/target from raw data.

Reply:
I mean, giving them control of getting the predicts/target from raw data and having these 4 additional eval func are different topics. Like we discussed, I will add the capability of inference by user for this custom eval func back.